webapp/components/CommentList/CommentList.vue

```diff
@@ -40,5 +40,12 @@ export default {
   props: {
     post: { type: Object, default: () => {} },
   },
+  methods: {
+    deleteComment(deleted) {
+      this.post.comments = this.post.comments.map(comment => {
+        return comment.id === deleted.id ? deleted : comment
+      })
+    },
+  },
```

Comment:
maybe this is what updates the list?

Reply:
Exactly. A deleted comment does not get replaced (you would use `Array.filter`) but replaced with the response of `DeleteComment` (use `Array.map` and replace when found). The server response is called `deleted`, this is the argument of the `this.$emit` call.

If find this technique a much better separation of responsibilities. One component does the mutation and hands over the server response to the component which renders the list.

Reply:
I like it as well, less code as well
Also, more consistency in the code base, how it is handled in one place, it is handled in another